### PR TITLE
Generalized HasProperties from openxml-docx

### PR DESCRIPTION
**Pull request by kobsy**
Expects properties to be in a `Properties` module that is a sibling to `Elements`.

### CODE REVIEW

File: lib/openxml/has_properties.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      def value_property(name, as: nil)
+        attr_reader name
+
+        properties[name] = (as || name).to_s
+
+        class_eval <<-CODE, __FILE__, __LINE__ + 1
+        def #{name}=(value)
+          property_key = "#{name}".to_sym
+          class_name = properties[property_key].split("_").map(&:capitalize).join
+          prop_module = class.name.split("::")[0...-2].push("Properties").join("::").constantize
```
Comment (boblail) [inline lib/openxml/has_properties.rb]: Is `.constantize` from ActiveSupport? (we don't list that as a runtime dependency of this library — and I'd like not to)

Comment (kobsy) [reply lib/openxml/has_properties.rb]: D'oh! I don't know if I realized that was from ActiveSupport. I'll refactor it away...

**Comment by kobsy** [comment]
Okay, ready for another look, @boblail

File: lib/openxml/has_properties.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      def value_property(name, as: nil)
+        attr_reader name
+
+        properties[name] = (as || name).to_s
+
+        class_eval <<-CODE, __FILE__, __LINE__ + 1
+        def #{name}=(value)
+          property_key = "#{name}".to_sym
+          class_name = properties[property_key].split("_").map(&:capitalize).join
+          prop_module = Object.const_get class.name.split("::")[0...-2].push("Properties").join("::")
```
Comment (boblail) [inline lib/openxml/has_properties.rb]: I wonder why Rails has a more doesn't just use `Object.const_get`

https://github.com/rails/rails/blob/v5.1.3/activesupport/lib/active_support/inflector/methods.rb#L258-L287

What are they working around? ....

Comment (kobsy) [reply lib/openxml/has_properties.rb]: My guess was that they were allowing for things other than fully-qualified names, e.g., `::ClassName` and the like...

**Pull request merged**